chalk-solve/src/rust_ir.rs

```diff
@@ -645,6 +645,8 @@ pub struct OpaqueTyDatum<I: Interner> {
 pub struct OpaqueTyDatumBound<I: Interner> {
     /// Trait bounds for the opaque type.
     pub bounds: Binders<Vec<QuantifiedWhereClause<I>>>,
+    /// Where clauses that inform well-formedness conditions for the opaque type.
```

Comment:
Clarify by adding: These are conditions on the generic parameters of an opaque type which must be true for a reference to the opaque type to be well-formed. (Is that indeed clearer? Maybe not.)